pkg/console/service/kubevirt-proxy_test.go

```diff
@@ -1,9 +1,221 @@
 package service
 
 import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+
 	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/emicklei/go-restful/v3"
+	"github.com/golang/mock/gomock"
+	authnv1 "k8s.io/api/authentication/v1"
+	authzv1 "k8s.io/api/authorization/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	k8stesting "k8s.io/client-go/testing"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	api "github.com/kubevirt/vm-console-proxy/api/v1alpha1"
 )
 
 var _ = Describe("Kubevirt proxy", func() {
-	// TODO: Add tests
+
+	const (
+		testNamespace = "test-namespace"
+		testName      = "test-name"
+	)
+
+	var (
+		testVm *kubevirtv1.VirtualMachine
+
+		apiClient   *fake.Clientset
+		virtClient  *kubecli.MockKubevirtClient
+		vmInterface *kubecli.MockVirtualMachineInterface
+
+		testService *kubevirtProxy
+
+		request  *restful.Request
+		response *restful.Response
+		recorder *httptest.ResponseRecorder
+	)
+
+	BeforeEach(func() {
+		apiClient = fake.NewSimpleClientset()
+
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubecli.NewMockKubevirtClient(ctrl)
+		virtClient.EXPECT().AuthenticationV1().Return(apiClient.AuthenticationV1()).AnyTimes()
+		virtClient.EXPECT().AuthorizationV1().Return(apiClient.AuthorizationV1()).AnyTimes()
+		virtClient.EXPECT().CoreV1().Return(apiClient.CoreV1()).AnyTimes()
+		virtClient.EXPECT().RbacV1().Return(apiClient.RbacV1()).AnyTimes()
+
+		testVm = &kubevirtv1.VirtualMachine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testName,
+				Namespace: testNamespace,
+			},
+			Spec: kubevirtv1.VirtualMachineSpec{},
+		}
+
+		vmInterface = kubecli.NewMockVirtualMachineInterface(ctrl)
+		vmInterface.EXPECT().Get(testName, gomock.Any()).DoAndReturn(
+			func(_ string, _ any) (*kubevirtv1.VirtualMachine, error) {
+				if testVm != nil {
+					return testVm, nil
+				}
+				return nil, errors.NewNotFound(kubevirtv1.Resource("virtualmachines"), testName)
+			},
+		).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachine(testNamespace).Return(vmInterface).AnyTimes()
+
+		testService = &kubevirtProxy{
+			kubevirtClient: virtClient,
+		}
+
+		request = restful.NewRequest(&http.Request{
+			Header: make(http.Header),
+		})
+		request.PathParameters()["namespace"] = testNamespace
+		request.PathParameters()["name"] = testName
+
+		recorder = httptest.NewRecorder()
+		response = restful.NewResponse(recorder)
+		response.SetRequestAccepts(restful.MIME_JSON)
+	})
+
+	Context("TokenHandler", func() {
```

Comment:
Is there a test to check what happens if we get empty/nil parameters, inside the handler?

Reply:
DOne